src/components/forms/DebouncedCombobox.tsx

```diff
@@ -281,31 +281,35 @@ export function DebouncedCombobox(props: Props) {
               onFocus={props.onInputFocus}
             />
 
-            {props.clearButton && !props.disabled && !isLoading && (
-              <div className="absolute inset-y-0 right-0 mt-2.5 mr-1 cursor-pointer">
-                <X
-                  className="absolute inset-y-0 right-0 text-gray-400 h-4"
-                  onClick={() => {
-                    setSelectedOption({
-                      record: internalRecord,
-                      withoutEvents: true,
-                    });
-
-                    props.onClearButtonClick();
-                  }}
-                />
-              </div>
-            )}
-
-            {!props.clearButton && !isLoading && (
-              <div className="absolute inset-y-0 right-0 mt-2.5 mr-1 cursor-pointer">
-                <ChevronDown
-                  className="absolute inset-y-0 right-0 text-gray-400 h-4"
-                  aria-hidden="true"
-                  onClick={() => openDropdownButton.current?.click()}
-                />
-              </div>
-            )}
+            {props.clearButton &&
+              (props.defaultValue || props.value) &&
+              !props.disabled &&
+              !isLoading && (
```

Comment:
Yeah, maybe not the clearest way to handle it, but I can extract the variable somehow if that's ok with you?

The point of this is to fix the X or DownArrow displaying error. If we have the `clearButton` property set to true, and if this component is `not disabled` and `isLoading` is false, we will get an X even if we have nothing selected. That's not good behavior. We should only show X if something is selected.

So my logic is to additionally add a condition to check the `defaultValue` or the `value` property, if one of those properties is true we'll show an X, otherwise we'll show a DownArrow, the condition for displaying DownArrow is just below this one.